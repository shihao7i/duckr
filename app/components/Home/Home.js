import React from 'react';
import { container, title, slogan } from './styles.css';
import { I18n } from 'react-i18next';

import i18next from 'config/i18n'; // initialized i18next instance using reactI18nextModule

export default function Home () {
  return (
    <I18n>
      {(t) => {
        return (
            <div className={container}>
              <p className={title}>{t('title.label')}</p>
              <p className={slogan}>{t('description.label')}</p>
              <button>{'Toggle Language'}</button>
            </div>
          );
      }}
    </I18n>
  );
}
