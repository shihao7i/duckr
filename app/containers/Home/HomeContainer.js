import React from 'react';
import { Home } from 'components';
import i18next from 'config/i18n';

class HomeContainer extends React.Component {
  componentDidMount() {
    document.querySelector('button').addEventListener('click', () => {
      if (i18next.language !== 'en') {
        i18next.changeLanguage('en');
      } else {
        i18next.changeLanguage('cn');
      }
    });
  }

  render() {
    return <Home />;
  }
}

export default HomeContainer;
