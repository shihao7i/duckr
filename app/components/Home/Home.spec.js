import React from 'react';
import ReactDOM from 'react-dom';
import Home from './Home.js';
import renderer from 'react-test-renderer';

describe('Authenticate', () => {
  it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<Home/>, div);
  });

  it('renders a snapshot', () => {
    const tree = renderer.create(<Home/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});