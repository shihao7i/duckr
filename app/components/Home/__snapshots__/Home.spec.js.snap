// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authenticate renders a snapshot 1`] = `
<div
  className="container"
>
  <p
    className="title"
  >
    Duckr
  </p>
  <p
    className="slogan"
  >
    The real time, cloud based, modular, scalable, growth hack, social platform. In the cloud.
  </p>
  <button>
    Toggle Language
  </button>
</div>
`;
